pkg/workceptor/workceptor.go

```diff
@@ -383,7 +385,22 @@ func (w *Workceptor) monitorRemoteStdout(ctx context.Context, cancel context.Can
 				logger.Debug("Could not open stdout file %s: %s\n", stdoutFilename, err)
 				continue
 			}
+			doneChan := make(chan struct{})
+			go func() {
+				select {
+				case <-doneChan:
+					return
+				case <-ctx.Done():
+					cr, ok := conn.(interface{ CancelRead() })
```

Comment:
this checks that the interface is defined for `conn`? Trying to understand when `ok` is false

Reply:
Since `conn` is a `net.Conn`, we only know that it has the functions defined in the `net.Conn` interface declaration, which don't include `CancelRead()`.  But I happen to know that this particular `net.Conn` is probably always going to be a `netceptor.Conn`, which _does_ have a `CancelRead()`.  So what I'm doing here is using an anonymous interface `interface { CancelRead() }` to get access to the `CancelRead()` method via type assertion.  Without the `ok` this would panic if the type assertion failed, and I don't want that - I want it to still try its best to work if you pass it something other than a `netceptor.Conn`.  So I use the two variable version of type assertion, where you get a boolean that tells you if the assertion worked.  If `ok` comes back `True` then we know it's okay to call `cr.CancelRead()`.  If it comes back `False` then we have a non-netceptor Conn and just have to hope it will figure out how to unblock itself when we call `Close()` on it.